files/system/usr/share/polkit-1/rules.d/usbguard.rules

```diff
@@ -0,0 +1,24 @@
+// Allow all users to view USBGuard rules
+
+polkit.addRule(function(action, subject) {
+    if ((action.id == "org.usbguard.Policy1.listRules" ||
+         action.id == "org.usbguard.Devices1.listDevices" ||
+         action.id == "org.usbguard1.getParameter" ||
+        subject.active == true && subject.local == true) {
+            return polkit.Result.YES;
+    }
+});
+
+// Allow privileged users to change rules by entering their user password
+// This does not require a wheel account
+
+polkit.addRule(function(action, subject) {
+    if ((action.id == "org.usbguard.Policy1.appendRule" ||
+         action.id == "org.usbguard.Policy1.removeRule" ||
+         action.id == "org.usbguard.Devices1.applyDevicePolicy" ||
+         action.id == "org.usbguard1.setParameter") &&
+        subject.active == true && subject.local == true && (
+        subject.isInGroup("wheel") || subject.isInGroup("usbguard")) {
```

Comment:
this isn't necessary if you just do 

`return polkit.Result.AUTH_ADMIN`

then this isn't limited to wheel users :)

Reply:
what would ADMIN imply then? I understood that it means the wheel users password

Reply:
@secretmango it implies that it will prompt for a wheel user's password. but any user can view and interact with that prompt

Reply:
and would this then be safe? The rule should only allow either wheel users (the default) or a user in the dedicated usbguard group to manage these.

Other users should not be able to change the rules.

AUTH_ADMIN tells me that a specific user needs to enter their password, in contrast to AUTH_SELF, which could only be allowed when the user is in that group

Reply:
I don't know what you mean. auth_admin prompts for an administrative user to authenticate. if they don't, then the action isn't permitted.

It seems like you have some confusion or misunderstanding about polkit. 

As an example, as a non-wheel user run `ujust set-kargs-hardening` or any other ujust command that requires `sudo`. polkit will prompt you to authenticate as a wheel user, even though you ran it as a non-wheel user.

Reply:
yes, that is clear. But what if there is no wheel user? Then AUTH_ADMIN in general makes no sense, instead it should require a user in a dedicated group like `usbguard` and AUTH_SELF there.

This is just planning ahead. Android works without root/wheel entirely for example. If we remove the need for a wheel user in all new implementations, it will become less of a burden to use a system without a wheel user.

Reply:
what if the user daily driving the system isn't a member of the usbguard group?

Reply:
adding a user to the usbguard group would be documented in the postinstall readme. By default in that case a wheel user would be allowed to authenticate. If not, a wheel user needs to do the changes.

I am not sure about GUI integration effects here.

Also, what happens if the usbguard init script is ran again?

I will test the "use a dedicated group and AUTH_SELF" soon and report if it works as expected"

Reply:
also, the problem first and foremost with desktop linux is the lack of a real boundary between root and non-root. simply moving operations from admin operations to non-admin operations undermines the reason we want that boundary to begin with.

there are many ways to make it easier to use a system without a wheel user, not all of them are good ideas.

You could make the entire system RW and grant all users permissions for all files. It would be a terrible idea, but it would make things a whole lot easier for non-wheel users :)

Reply:
> I dont understand the point where you describe how a system could just allow all access

It was hyperbole for the sake of making a broader point

> So a goal should be to avoid needing to login as a wheel user entirely, shouldnt it?

AUTH_ADMIN doesn't require a wheel session. you are misunderstanding something still.

Reply:
Let's just keep this like you have it but add the usbguard group membership to the ujust command instead, so people don't have to read the docs to get added. They can just rerun the ujust command.

Hopefully that's a reasonable solution :)

Reply:
A disadvantage would be that a user needs to log out. I would change it to AUTH_ADMIN then, thanks!